plugin.rb

```diff
@@ -5,62 +5,81 @@
 # version: 0.1
 # authors: Neil Lalonde
 # url: https://github.com/discourse/discourse-saved-searches
+# transpile_js: true
 
 enabled_site_setting :saved_searches_enabled
 
 register_asset 'stylesheets/saved-searches.scss'
 
 after_initialize do
-
-  plugin = self
-
-  require_dependency File.expand_path('../app/jobs/regular/saved_search_notification.rb', __FILE__)
-  require_dependency File.expand_path('../app/jobs/scheduled/schedule_saved_searches.rb', __FILE__)
-
-  User.register_custom_field_type('saved_searches', :json)
-
-  add_to_serializer(:user, :saved_searches, false) do
-    (object.custom_fields['saved_searches'] || {})['searches']
-  end
-
-  add_to_serializer(:user, :include_saved_searches?, false) do
-    plugin.enabled? && scope.can_edit?(object)
-  end
-
   module ::SavedSearches
     class Engine < ::Rails::Engine
       engine_name 'saved_searches'
       isolate_namespace SavedSearches
     end
   end
 
-  require_dependency 'application_controller'
-
-  class ::SavedSearches::SavedSearchesController < ::ApplicationController
-    requires_plugin 'discourse-saved-searches'
+  require File.expand_path('../app/controllers/saved_searches_controller.rb', __FILE__)
+  require File.expand_path('../app/jobs/regular/execute_saved_searches.rb', __FILE__)
+  require File.expand_path('../app/jobs/scheduled/schedule_saved_searches.rb', __FILE__)
+  require File.expand_path('../app/models/saved_search_result.rb', __FILE__)
+  require File.expand_path('../app/models/saved_search.rb', __FILE__)
+  require File.expand_path('../app/serializers/saved_search_serializer.rb', __FILE__)
+  require File.expand_path('../lib/email_user_extensions.rb', __FILE__)
+  require File.expand_path('../lib/guardian_extensions.rb', __FILE__)
+  require File.expand_path('../lib/user_extensions.rb', __FILE__)
+  require File.expand_path('../lib/user_notifications_extensions.rb', __FILE__)
+
+  NotificationEmailer::EmailUser.class_eval { prepend EmailUserExtensions }
+  Guardian.class_eval { prepend GuardianExtensions }
+  User.class_eval { prepend UserExtensions }
+  UserNotifications.class_eval { prepend UserNotificationsExtensions }
```

Comment:
```suggestion
  reloadable_patch do
    NotificationEmailer::EmailUser.class_eval { prepend EmailUserExtensions }
    Guardian.class_eval { prepend GuardianExtensions }
    User.class_eval { prepend UserExtensions }
    UserNotifications.class_eval { prepend UserNotificationsExtensions }
  end
```
This is required so that these patches are re-applied after a zeitwork reload.

(You can repro the issue by starting the server, saving a change to a random core autoloaded file like `posts_controller.rb`, and then refresh the user preferences page. It will 404 due to a missing guardian method)

Reply:
That did not happen for me while developing the plugin, but it makes sense, so I'll add this suggestion. Thanks!